src/interactive-window/interactiveWindowProvider.ts

```diff
@@ -132,6 +156,15 @@ export class InteractiveWindowProvider
         if (!result) {
             // No match. Create a new item.
             result = await this.create(resource, mode, connection);
+            // We already check preferred kernel before creating the interactive window
+            // no need to update it anymore.
+            await result.start(undefined);
+        } else {
+            const preferredController = connection
+                ? this.controllerRegistration.get(connection, InteractiveWindowView)
+                : await this.controllerDefaultService.computeDefaultController(resource, InteractiveWindowView);
+
+            await result.start(preferredController);
```

Comment:
I think we should let the `create` method figure out the preferred controller and pass that into the contructor.
that way we don't have two ways of starting the IW with & without arguments for the notebook controller.
Also cleaner, else the controller can be changed from outside, when i think it should be a value passed into the ctor at the point of creating the IW.

Basically i think we can move this `preferedController = connection...` code into the `create` method & then remove the public method `start`

Reply:
+1, agree with Don. Controller in the constructor makes the logic simpler in the class itself.

Reply:
> I think we should let the create method figure out the preferred controller and pass that into the contructor.
that way we don't have two ways of starting the IW with & without arguments for the notebook controller

This is what `InteractiveWindowProvider#create` is already doing now, it creates controller based on connection first and then create `InteractiveWindow` objects. While on window reload, we create `InteractiveWindow` from cache, at which point we are not in this code path of `InteractiveWindowProvider#create` but in a `get` path.